g2p/deprecation.py

```diff
@@ -0,0 +1,32 @@
+"""Hide deprecation and version checking code here so it's not in the way.
+
+Usage: g2p.__init__ should simply do "import g2p.deprecation" in order to get the Python
+version validated. Other functions can be added and invoked as needed.
+"""
+
+import sys
+
+from g2p.log import LOGGER
+import g2p._version
+
+if sys.version_info < (3, 6):  # pragma: no cover
+    sys.exit(
+        f"Python 3.6 or more recent is required by g2p. You are using {sys.version}.\n"
+        "Please use a newer version of Python."
+    )
+
+
+def handle_tok_lang_deprecation(tok_lang):
+    """Warn or raise about using the deprecated tok_lang arg to make_g2p"""
+    if tok_lang:
+        if g2p._version.VERSION < "2.0":
+            LOGGER.warning(
```

Comment:
Should we be using the `warnings` module instead?  See example: https://docs.python.org/3/library/warnings.html#testing-warnings

Reply:
I'm not familiar with that module, and it's not super clear from the documentation how to use it. Do you have an example of what that would look like in our code, and would it be displayed any time someone use g2p calling make_g2p with tok_langs?

Reply:
You can use `logging.captureWarnings` apparently: https://docs.python.org/3/library/logging.html#logging.captureWarnings

There are some other options too though it looks a bit complicated:

https://docs.python.org/3/library/warnings.html#describing-warning-filters

Reply:
Right, I just managed to get some output when I add this:

    warnings.filterwarnings("default", category=DeprecationWarning)

and if I adjust my stacklevel to 3, I get this output when I revert g2p.cli to use tok_langs:

```
$ g2p convert pelle fra fra-ipa
C:\Users\joanise\Miniconda3\envs\ras-py38\lib\site-packages\eventlet\support\greenlets.py:6: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
  preserves_excinfo = (distutils.version.LooseVersion(greenlet.__version__)
C:\Users\joanise\Miniconda3\envs\ras-py38\lib\site-packages\eventlet\support\greenlets.py:7: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
  >= distutils.version.LooseVersion('0.3.2'))
pʌl
c:\users\joanise\sandboxes\g2p\g2p\cli.py:563: DeprecationWarning: Deprecation warning: the tok_lang argument to make_g2p is deprecated, and will be removed in g2p version 2.0 Use tokenize=True or create a custom_tokenizer using make_tokenizer() instead.
  transducer = make_g2p(
```

Reply:
I'm not keen on seeing the eventlet deprecation warning, though, I'll want to figure out how to only activate our own.